src/tools/crawling_tools.py

```
@@ -516,132 +652,201 @@ async def crawl_dir(
+                file_results = await with_timeout(
+                    crawl_directory(
+                        directory_path=directory_path,
+                        max_files=max_files,
+                        max_file_size=max_file_size,
+                        exclude_patterns=exclude_patterns or [],
+                        include_patterns=include_patterns
+                    ),
+                    timeout_seconds=TimeoutConfig.CRAWLER_DIRECTORY_TIMEOUT,
+                    operation_name=f"crawl directory {directory_path}"
+                )
+            except asyncio.TimeoutError as timeout_error:
```

Ruff (0.12.2)

Replace aliased errors with `TimeoutError`

Replace `asyncio.TimeoutError` with builtin `TimeoutError`

(UP041)